nebullvm/operations/optimizations/compilers/openvino.py

```diff
@@ -25,6 +30,132 @@
 from nebullvm.tools.transformations import MultiStageTransformation
 
 
+
+class TensorFlowOpenVINOCompiler(Compiler):
+    supported_ops = {
+        "cpu": [
+            None,
+            # QuantizationType.STATIC,
+            # QuantizationType.HALF,
+        ],
+        "gpu": [],
+    }
+
+    def __init__(self):
+        super().__init__()
+
+    def execute(
+        self,
+        model: Model,
+        model_params: ModelParams,
+        temp_file_path: Path,
+        input_tfms: MultiStageTransformation = None,
+        metric_drop_ths: float = None,
+        quantization_type: QuantizationType = None,
+        input_data: DataManager = None,
+        **kwargs,
+    ):
+        """Compile the input model using TF-OPENVINO library.
+
+        Args:
+            model (tf.keras.Model()): The tensorflow model.
+            model_params (ModelParams): The model parameters.
+            output_file_path (str or Path): Path where storing the output
+            TensorFlow file.
+            input_tfms (MultiStageTransformation, optional): Transformations
+                to be performed to the model's input tensors in order to
+                get the prediction. Default: None.
+            metric_drop_ths (float, optional): Threshold for the accepted drop
+                in terms of precision. Any optimized model with a higher drop
+                will be ignored. Default: None.
+            quantization_type (QuantizationType, optional): The desired
+                quantization algorithm to be used. Default: None.
+            input_data (DataManager): User defined data. Default: None
+        """
+
+        if quantization_type not in self.supported_ops[self.device.value]:
+            self.compiled_model = None
+            return
+
+        if quantization_type is QuantizationType.STATIC and input_data is None:
+            raise ValueError("Input data is required for static quantization.")
+
+        self.logger.info(
+            f"Optimizing with {self.__class__.__name__} and "
+            f"q_type: {quantization_type}."
+        )
+
+        check_quantization(quantization_type, metric_drop_ths)
+        train_input_data = input_data.get_split("train").get_numpy_list(
+            QUANTIZATION_DATA_NUM
+        )
+
+
+        # save model to temporary folder
+        dirpath = tempfile.mkdtemp()
```

Comment:
as temporary directory you can use the directory that is passed to each compiler execute with the parameter `onnx_output_path`, you can find it in `nebullvm/operations/optimizations/base.py` when `compiler_op.to(self.device).execute` is called. I would also rename the parameter to something more general (such as  `tmp_dir_path`), and when you rename it make sure to rename it wherever it's used.